### ISSUE TITLE
Halcyon 9.41.7023

### ISSUE THREAD

**Issue by Ana-Green** [HalcyonGrid/halcyon#89]
TP to region, load raw terrain, regions land, and water start to bump, moves up and down if I relog sometimes even 3  times its gone [ Bug ].

Replicate: Goto one of your regions [ Empty one ] load raw terrain, regions land, and water starts to bump, Solution relog.  

Regards,
Ana Green

**Comment by emperorstarfinder** [comment]
The fix is to simply do a region restart of the region after loading the terrain.  That will force send an update to the rest of the services that the terrain has changed and your viewer will be fine in that region after.

Tested via our internal Halcyon test bed grid

**Comment by Ana-Green** [comment]
Yes, I know that! So you call that "A fix?" Let me give you an example, John pays for his GOR region, he has 10 minutes to change the rock formation so he does so and also loads a new terrain.raw file. But then all his members start to jump - raiders on the dock can also not stand still. But John knows the fix, "he has to restart the complete region with that ...  all either leave or get kicked off, anyway due to the restart, this is rather comical than a Fix. It's not even a Fix, or is it too simple that it for you to complicated to even Fix it? 

It looks, I did not say it is,  but it looks like a collision because, in fact, you start to bump up and down. Even OpenSim does not have that nor does Second Life because I saw it, Linden Labs loaded a new terrain for my region and nothing started to Jojo bump, jump ridiculously.  It's a bad code (Physics area?) hence a bug.
Don't forget Halcyon is inherited from OpenSim code what did not have this weird effect. I might be not Dad but  I am also not a dumb girl.  What's the point to report a new issue here. I am sure Justin would have agreed with me and fixed it. Do I agree with your Fix?  Nope.

What a Shame, Regards,

Work for Intel and CISCO, from the USA, 27 years of age, SL/LL BetaGrid Tester, Firestorm Beta Tester. started at 9/12/2018 at NASA. 
AFNORTH/AWACS Brunssum.
Ana Green

**Comment by emperorstarfinder** [comment]
Is it a permanent fix? No, it isn't.  But it does fix the problem at least until it can be fixed in the code,

It is not necessarily a collision.  The reason your seeing that is because unlike OpenSim (and many of its derivatives, Halcyon has a problem with how the terrain's maps are updated meaning they don't update in real time which they should be when there is a new terrain loaded.

By restarting the region (and I should have been a bit more clear last night in the steps), it will make sure the region knows about the updated terrain maps (i.e. the terrain's new heightmap, etc.) and will send the correct info to both the grid services and the viewer.  Thus causing the shaking you are seeing to go away when you load a new terrain file into a region.

The correct way to fix this (which is the simple fix or a workaround until it can be fixed in halcyon's codebase) is as follows:

1. In the console that runs the region with the changed terrain press Ctrl + C to shut the region console down.

2. When asked if you are sure you want to shut the console down (or terminate the console) say Y and hit enter

3. After the region's console has closed you can simply launch the console again as normal.

4. This will cause the region to be reloaded with the updated information including the new information for the terrain.

5. Teleport to the region once it is back online and the shaking will be gone.

Is it an inconvenience? Of course, it can be, but then sometimes workarounds can be until a more permanent fix can be had.  It has nothing to do with OpenSim currently or any of their developers, nor does it have to do with whether your stupid or smart or where you work.  A suggested workaround has been provided.  It is up to you whether you try it and see for yourself.  At least if the workaround works for you it is a temporary fix that allows you to move forward until halcyon's devs can find a more permanent fix to the problem in the code.

**Comment by Ana-Green** [comment]
Thank you for your time, well I saw it I did report it and yes I can live with a temporary fix.

Regards,
Work for Intel and CISCO, from the USA, 27 years of age, SL/LL BetaGrid Tester, Firestorm Beta Tester. started at 9/12/2018 at NASA.
AFNORTH/AWACS Brunssum.
Ana Green
